src/compiler.ts

```diff
@@ -6340,6 +6349,7 @@ export class Compiler extends DiagnosticEmitter {
               nativeFieldType, field.memoryOffset
             ),
             fieldType,
+            fieldType, // TODOFIX
```

Comment:
Wouldn't this become `valueType` for `valueExpr` (newExpr in makeReplace) and `fieldType` for the generated load (oldExpr in makeReplace)?